modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -799,48 +746,70 @@ def _take_2d_positional(
         if col_positions is None and row_positions is None:
             return self.copy()
 
-        sorted_row_positions = sorted_col_positions = None
+        import pandas._libs.internals as libinternals
 
         if row_positions is not None:
-            sorted_row_positions = self._get_sorted_positions(row_positions)
-            # Get dict of row_parts as {row_index: row_internal_indices}
-            row_partitions_dict = self._get_dict_of_block_index(
-                0, sorted_row_positions, are_indices_sorted=True
-            )
-            new_row_lengths = self._get_new_lengths(row_partitions_dict, axis=0)
-            new_index, _ = self._get_new_index_obj(
-                row_positions, sorted_row_positions, axis=0
-            )
+            row_positions = np.asarray(row_positions, dtype=np.intp)
```

Comment:
@jbrockmendel one of the current errors in CI is the loss of metadata, in particular `freq` field. This is because you are converting Range object to numpy array in this line.